### ISSUE TITLE
Provide benchmarks for each release

### ISSUE THREAD

**Issue by espetro** [dry-python/returns#826]
Hi there,

i love the work you’re doing with `returns`. Perhaps one of the main downsides of applying functional type classes and data types in interpreted languages is that probably you’re worsening your application performance. Do we already have any kind of benchmark to check if this holds?

If `returns` doesn’t have such a tool, I’d love to contribute to it.

**Comment by sobolevn** [comment]
@espetro it would be awesome to have such a tool! PR is welcome!

**Comment by thepabloaguilar** [comment]
Maybe we can use of theses libs:
* [pytest-benchmark](https://github.com/ionelmc/pytest-benchmark)
* [pyperformance](https://pyperformance.readthedocs.io/index.html)

**Comment by espetro** [comment]
Hi, @thepabloaguilar @sobolevn are you aware if we have such tooling in place already? Otherwise I'd look into `codespeed` as @thepabloaguilar shared.

**Comment by sobolevn** [comment]
Yes, I am using codspeed for `django-modern-rest` already: https://github.com/wemake-services/django-modern-rest/

Feel free to send a PR :)

**Comment by thepabloaguilar** [comment]
@espetro I've set up the codspeed integration, you can check it [here](https://app.codspeed.io/dry-python/returns)

Feel free to open a new PR to add more benchmarks tests [here](https://github.com/dry-python/returns/blob/master/benchmarks/test_benchmarks.py)
